### PR TITLE
New Feed Events

### PR DESCRIPTION
**Pull request by cbarrazaa1**
- Removed properties from CheckInEvent because we will move those properties into their own individual FeedEvents. We'll keep the CheckInEvent class even though it's not different at all from FeedEvent just in case we need to add new properties to checkin events.
- Added new classes for the photo and review addition feed events.

### CODE REVIEW

File: Models/Feed/PhotoAdditionEvent.m [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by César Francisco Barraza on 8/1/18.
+//  Copyright © 2018 Bevin Benson. All rights reserved.
+//
+
+#import "PhotoAdditionEvent.h"
+
+@implementation PhotoAdditionEvent
+- (instancetype)initWithParseObject:(PFObject *)object {
+    self = [super initWithParseObject:object];
```
Comment (stephinphection) [inline Models/Feed/PhotoAdditionEvent.m]: use the same 

```
if (self = [super initWithParseObject:object]) {
    self.photos = object[@"photos"];
}

return self;
```

File: Models/Feed/ReviewAdditionEvent.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+//  ReviewAdditionEvent.h
+//  social-maps
+//
+//  Created by César Francisco Barraza on 8/1/18.
+//  Copyright © 2018 Bevin Benson. All rights reserved.
+//
+
+#import "FeedEvent.h"
+
+@interface ReviewAdditionEvent : FeedEvent
```
Comment (stephinphection) [inline Models/Feed/ReviewAdditionEvent.h]: same feedback as PhotoAdditionEvent.m

File: Models/Feed/ReviewAdditionEvent.m [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//
+//  Created by César Francisco Barraza on 8/1/18.
+//  Copyright © 2018 Bevin Benson. All rights reserved.
+//
+
+#import "ReviewAdditionEvent.h"
+
+@implementation ReviewAdditionEvent
+- (instancetype)initWithParseObject:(PFObject *)object {
+    self = [super initWithParseObject:object];
```
Comment (stephinphection) [inline Models/Feed/ReviewAdditionEvent.m]: same feedback as PhotoAdditionEvent.m

**Comment by stephinphection** [comment]
documentation explaining the lil nits https://developer.apple.com/documentation/objectivec/nsobject/1418641-init?language=objc


PS I think I missed this on a previous PR i didn't read thoroughly enough - my bad!

**Comment by cbarrazaa1** [comment]
Forgot to do this fix in the FeedEvent and CheckInEvent classes, just pushed those too.

File: Models/Feed/CheckInEvent.m [hunk 4]
```diff
@@ -10,16 +10,15 @@
 
 @implementation CheckInEvent
 - (instancetype)initWithParseObject:(PFObject *)object {
```
Comment (stephinphection) [inline Models/Feed/CheckInEvent.m]: if you're not doing any additional initialization here, I don't think you even need to override genTabContentListViewSurface

File: Models/Feed/FeedEvent.m [hunk 5]
```diff
@@ -12,15 +12,16 @@
 
 @implementation FeedEvent
 - (instancetype)initWithParseObject:(PFObject *)object {
-    self = [super init];
-    self.parseObject = object;
-    self.user = object[@"user"];
-    self.place = object[@"place"];
-    self.eventType = [object[@"eventType"] unsignedIntegerValue];
+    if(self = [super init])
```
Comment (stephinphection) [inline Models/Feed/FeedEvent.m]: perfectooooo

**Pull request merged**